### PR TITLE
`--init_package` flag

### PR DESCRIPTION
**Pull request by ZanSara**
See #10. Add feature, tests and some documentation in README.md

### CODE REVIEW

**Comment by ZanSara** [comment]
Having started from my previous PR branch, there are a lot of unrelated commits here. Please process #3 first.

File: README.md [hunk 1]
```diff
@@ -93,6 +93,7 @@ Whether running via the command line or from a script, the arguments and options
 * **rccOptions** - Additional options to pass to the resource compiler. See the man page of pyrcc5 for more information on options. An example of a valid option would be "-compress 1". Default is to pass no options.
 * **uicOptions** - Additional options to pass to the UI compiler. See the man page of pyuic5 for more information on options. An example of a valid option would be '--from-imports'. Default is to pass no options.
 * **force** - Specifies whether to force compile all of the files found. The default is false meaning only outdated files will be compiled.
+* **init_package** - Specifies whether to check that the folder containing the generated files should also contain an `__init__.py` file. Default is True meaning the (empty) file will be added if missing.
```
Comment (zkovari) [inline README.md]: this is a bit difficult to understand. Could we simplify it something like "If specified, an empty `__init__.py` file is also generated in the output directory. Default value is True"

File: pyqt5ac.py [hunk 2]
```diff
@@ -139,7 +142,7 @@ def replaceVariables(variables_definition, string_with_variables):
     return string_with_variables
 
 
-def main(rccOptions='', uicOptions='', force=False, config='', ioPaths=(), variables=None):
+def main(rccOptions='', uicOptions='', force=False, initPackage=True, config='', ioPaths=(), variables=None):
```
Comment (zkovari) [inline pyqt5ac.py]: why not to add the new argument to the end?

File: tests/test_pyqt5ac.py [hunk 3]
```diff
@@ -34,6 +38,21 @@ def _write_config_file(dir):
     return config
 
 
+def _write_config_file_without_init(dir):
```
Comment (zkovari) [inline tests/test_pyqt5ac.py]: I think it's okay if you remove this and you rather use the initPackage argument in the main() method

File: tests/test_pyqt5ac.py [hunk 4]
```diff
@@ -262,3 +281,48 @@ def test_ui_generation_with_variables(tmpdir):
 
     _assert_path_exists(tmpdir.join("generated"))
     _assert_path_exists(tmpdir.join("generated/main_ui.py"))
+
+
+def test_init_is_created(tmpdir):
```
Comment (zkovari) [inline tests/test_pyqt5ac.py]: you could remove this test case and rather extend the other valid generations (ui+resource) and assert there whether an extra __init__ file was created

File: tests/test_pyqt5ac.py [hunk 5]
```diff
@@ -262,3 +281,48 @@ def test_ui_generation_with_variables(tmpdir):
...
+
+def test_dont_check_for_init(tmpdir):
+    config = _write_config_file_without_init(tmpdir)
+    ui_file = tmpdir.mkdir("gui").join("main.ui")
+    _write_ui_file(ui_file)
+
+    pyqt5ac.main(config=str(config))
+
+    _assert_path_exists(tmpdir.join("generated"))
+    dest_file = tmpdir.join("generated/main_ui.py")
```
Comment (zkovari) [inline tests/test_pyqt5ac.py]: no need for this variable

File: tests/test_pyqt5ac.py [hunk 6]
```diff
@@ -262,3 +281,48 @@ def test_ui_generation_with_variables(tmpdir):
...
+    _assert_path_exists(dest_file)
+    init_file = tmpdir.join("generated/__init__.py")
```
Comment (zkovari) [inline tests/test_pyqt5ac.py]: no need for this variable

File: tests/test_pyqt5ac.py [hunk 7]
```diff
@@ -262,3 +281,48 @@ def test_ui_generation_with_variables(tmpdir):
...
+    config = _write_config_file(tmpdir)
+    ui_file = tmpdir.mkdir("gui").join("main.ui")
+    _write_ui_file(ui_file)
+    init_file = tmpdir.mkdir("generated").join("__init__.py")
+    init_file.write("test")
+
+    pyqt5ac.main(config=str(config))
+
+    _assert_path_exists(tmpdir.join("generated"))
+    dest_file = tmpdir.join("generated/main_ui.py")
```
Comment (zkovari) [inline tests/test_pyqt5ac.py]: no need for this variable

File: tests/test_pyqt5ac.py [hunk 8]
```diff
@@ -262,3 +281,48 @@ def test_ui_generation_with_variables(tmpdir):
...
+
+def test_init_is_created(tmpdir):
+    config = _write_config_file(tmpdir)
+    ui_file = tmpdir.mkdir("gui").join("main.ui")
+    _write_ui_file(ui_file)
+
+    pyqt5ac.main(config=str(config))
+
+    _assert_path_exists(tmpdir.join("generated"))
+    dest_file = tmpdir.join("generated/main_ui.py")
```
Comment (zkovari) [inline tests/test_pyqt5ac.py]: no need for this variable

File: tests/test_pyqt5ac.py [hunk 9]
```diff
@@ -262,3 +281,48 @@ def test_ui_generation_with_variables(tmpdir):
...
+    _assert_path_exists(dest_file)
+    init_file = tmpdir.join("generated/__init__.py")
```
Comment (zkovari) [inline tests/test_pyqt5ac.py]: no need for this variable

File: tests/test_pyqt5ac.py [hunk 10]
```diff
@@ -262,3 +281,48 @@ def test_ui_generation_with_variables(tmpdir):
...
+    _assert_path_exists(init_file)
```
Comment (zkovari) [inline tests/test_pyqt5ac.py]: you could also check that the content is empty

File: pyqt5ac.py [hunk 11]
```diff
@@ -72,9 +72,12 @@ def _isOutdated(src, dst, isQRCFile):
 @click.option('--config', '-c', default='', type=click.Path(exists=True, file_okay=True, dir_okay=False),
               help='JSON or YAML file containing the configuration parameters')
 @click.option('--force', default=False, is_flag=True, help='Compile all files regardless of last modification time')
+@click.option('--init-package', 'initPackage', default=True, is_flag=True,
+              help='Ensures that the folder containing the generated files is a Python submodule '
```
Comment (zkovari) [inline pyqt5ac.py]:
```suggestion
              help='Ensures that the folder containing the generated files is a Python subpackage '
```

no?

File: README.md (hunk 1 above)
Comment (zkovari) [reply README.md]: or sth similar that you had on the cli

**Review by zkovari — CHANGES_REQUESTED** [review]

**Comment by zkovari** [comment]
could you also resolve the conflict please?

File: pyqt5ac.py (hunk 2 above)
Comment (ZanSara) [reply pyqt5ac.py]: For similarity to the `cli()` signature, where it had no default value. In Python args without default must come first. However, to the end it goes (with True as default in both signatures).

Side note: Click was setting the default anyway to `True` right before calling `cli()`, so it really makes no difference.

Comment (zkovari) [reply pyqt5ac.py]: I prefer it with default value.

The order does matter if someone is calling this main method with positional arguments. And people do that actually, I made a little code search on GitHub last time.

**Review by zkovari — APPROVED** [review]

**Comment by zkovari** [comment]
@ZanSara thanks a lot!

**Pull request merged**